src/Analyzers/CSharp/Analyzers/RemoveUnnecessarySuppressions/CSharpRemoveUnnecessaryPragmaSuppressionsDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Reflection;
+using Microsoft.CodeAnalysis.CSharp.LanguageServices;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.RemoveUnnecessarySuppressions;
+
+namespace Microsoft.CodeAnalysis.CSharp.RemoveUnnecessarySuppressions
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class CSharpRemoveUnnecessaryPragmaSuppressionsDiagnosticAnalyzer
+        : AbstractRemoveUnnecessaryPragmaSuppressionsDiagnosticAnalyzer
+    {
+        protected override string CompilerErrorCodePrefix => "CS";
```

Comment:
so, woudl we not let you know about unnecessary IDEXXXX suppressions?  seems unfortunate.  i would have thought telling people about all unnecessary suppressions woudl be desirable.

Reply:
No, that is also supported. However, we need to treat compiler diagnostics specially. More specifically, compiler diagnostics can have pragma suppressions with just with integer ID, so we need this differentiation.

Reply:
ah, please doc somewhere (if not already doc'ed).

Reply:
Its documented in the base type in `IsSupportedId` method that uses this property.